src/action/base/create_directory.rs

```diff
@@ -236,6 +289,48 @@ impl Action for CreateDirectory {
     }
 }
 
+// There are cleaner ways of doing this (eg `systemctl status $PATH`) however we need a widely supported way.
+async fn path_is_mountpoint(path: &Path) -> Result<bool, ActionErrorKind> {
+    let path_str = match path.to_str() {
+        Some(path_str) => path_str,
+        None => return Ok(false),
```

Comment:
I don't think this should be `Ok` -- it means UTF8 validation failed, so we don't actually check if the path is a mountpoint because of that.